.github/workflows/validate.yaml

```diff
@@ -62,6 +66,15 @@ jobs:
         working-directory: ./src/github.com/${{ github.repository }}
         run: make generated-files
 
+      # Generate catalog only if not triggered by a pull request. Credentials are not available in pre-submit phase.
+      - name: Regenerate file-based catalog
+        if: github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || ( github.event_name == 'push' && !contains(github.ref_name, 'dependabot/') )
+        working-directory: ./src/github.com/${{ github.repository }}
+        env:
+          REGISTRY_REDHAT_IO_USERNAME: ${{ secrets.REGISTRY_REDHAT_IO_USERNAME }}
+          REGISTRY_REDHAT_IO_PASSWORD: ${{ secrets.REGISTRY_REDHAT_IO_PASSWORD }}
+        run: make generate-catalog
```

Comment:
I think we want to also run this on `bump` https://github.com/openshift-knative/serverless-operator/blob/da827643dd892789387b2c58b61e9ed97c2d907f/.github/workflows/bump.yaml#L91-L93